iac/network/scripts/bootnode_startup.sh

```diff
@@ -0,0 +1,160 @@
+sudo systemctl start docker
+sudo systemctl enable docker
+sudo usermod -aG docker $SSH_USER
+sudo systemctl restart docker
+
+# Ensure Docker starts on reboot
+echo "Configuring Docker to start on boot..."
+sudo systemctl enable docker.service
+sudo systemctl enable containerd.service
+
+if [ "$LOCATION" = "GCP" ]; then
+# Install Google Cloud Ops Agent for logging
+echo "Installing Google Cloud Ops Agent..."
+curl -sSO https://dl.google.com/cloudagents/add-google-cloud-ops-agent-repo.sh
+sudo bash add-google-cloud-ops-agent-repo.sh --also-install
+
+# Cloud Logging Configuration (Ops Agent)
+echo "Configuring Cloud Logging with Ops Agent..."
+cat <<EOF | sudo tee /etc/google-cloud-ops-agent/config.yaml
+logging:
+  receivers:
+    docker_logs:
+      type: files
+      include_paths:
+        - /var/lib/docker/containers/*/*.log
+
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [docker_logs]
+
+
+metrics:
+  receivers:
+    hostmetrics:
+      type: hostmetrics
+  service:
+    pipelines:
+      default_pipeline:
+        receivers: [hostmetrics]
+EOF
+
+# Enable and start Ops Agent
+sudo systemctl enable google-cloud-ops-agent
+sudo systemctl restart google-cloud-ops-agent
+elif [ "$LOCATION" = "AWS" ]; then
+  echo "Placeholder for AWS instances"
+fi
+
+# Create a start script to be run by systemd
+CONTAINER_NAME="aztec-bootnode"
+REPO=aztecprotocol
+IMAGE=aztec
+LOG_LEVEL=verbose
+
+cat <<EOF > /home/$SSH_USER/tag.sh
+#!/bin/bash
+export TAG=$TAG
+EOF
+chmod +x /home/$SSH_USER/tag.sh
+
+cat << 'EOF' > /home/$SSH_USER/start.sh
+#!/bin/bash
+source ./tag.sh
+echo "Starting bootnode container..."
+JSON=$(curl -s http://static.aztec.network/$NETWORK_NAME/bootnodes.json)
+export BOOTSTRAP_NODES=$(echo "$JSON" | jq -r '.bootnodes | join(",")')
+echo "Bootnode enrs: $BOOTSTRAP_NODES"
+docker pull $REPO/$IMAGE:$TAG
+docker run \
+ --restart=always \
+ --name $CONTAINER_NAME \
+ --volume $DATA_DIRECTORY:$DATA_DIRECTORY \
+ --publish $AZTEC_PORT:$AZTEC_PORT \
+ --publish $P2P_PORT:$P2P_PORT/udp \
+ --env DATA_DIRECTORY \
+ --env DATA_STORE_MAP_SIZE_KB \
+ --env P2P_UDP_ANNOUNCE_ADDR \
+ --env P2P_UDP_LISTEN_ADDR \
+ --env PEER_ID_PRIVATE_KEY \
+ --env L1_CHAIN_ID \
+ --env AZTEC_PORT \
+ --env BOOTSTRAP_NODES \
+ --env LOG_LEVEL \
+ $REPO/$IMAGE:$TAG node --no-warnings /usr/src/yarn-project/aztec/dest/bin/index.js start --p2p-bootstrap
+EOF
+chmod +x /home/$SSH_USER/start.sh
+
+
+DATA_DIRECTORY=/home/$SSH_USER/data
+
+P2P_UDP_ANNOUNCE_ADDR="$PUBLIC_IP:$P2P_PORT"
+P2P_UDP_LISTEN_ADDR="0.0.0.0:$P2P_PORT"
+
+mkdir -p $DATA_DIRECTORY
+
+# Create systemd service for bootnode
+echo "Creating systemd service to ensure container runs on startup with environment variables..."
+cat <<EOF | sudo tee /etc/systemd/system/aztec-bootnode.service
+[Unit]
+Description=Aztec Bootnode Service
+Requires=docker.service
+After=docker.service
+
+[Service]
+Restart=always
+RestartSec=10
+WorkingDirectory=/home/$SSH_USER
```

Comment:
Who's SSH user in practice? Is it a shared user between whoever can (re)deploy, or specific users? If it's the latter, it can be a pain to maintain.

Reply:
Ah, I see the `deploy_bootnodes` script creates an ssh key, I assume it's that one that gets used, so we're good.